xml/x509_subjects.xml

```diff
@@ -0,0 +1,124 @@
+    <param pos="0" name="os.product" value="Dell Remote Access Controller"/>
+    <param pos="0" name="os.device" value="BMC"/>
+    <param pos="0" name="service.vendor" value="Dell"/>
+    <param pos="0" name="service.family" value="Remote Access Controller"/>
+    <param pos="1" name="dell.service_tag"/>
+  </fingerprint>
+
+  <!-- iDRAC without valid service tags -->
+  <fingerprint pattern="^CN=idrac.*,OU=Remote Access Group,O=Dell Inc\.,L=Round Rock,ST=Texas,C=US$">
+    <description>Dell iDRAC Remote Access Controller</description>
+    <example>CN=idrac-SVCTAG,OU=Remote Access Group,O=Dell Inc.,L=Round Rock,ST=Texas,C=US</example>
+    <example>CN=idrac-prosit-laks,OU=Remote Access Group,O=Dell Inc.,L=Round Rock,ST=Texas,C=US</example>
+    <example>CN=idrac-,OU=Remote Access Group,O=Dell Inc.,L=Round Rock,ST=Texas,C=US</example>
+    <example>CN=idrac-SVCTAG,OU=Remote Access Group,O=Dell Inc.,L=Round Rock,ST=Texas,C=US</example>
+    <example>CN=idrac,OU=Remote Access Group,O=Dell Inc.,L=Round Rock,ST=Texas,C=US</example>
+    <param pos="0" name="os.vendor" value="Dell"/>
+    <param pos="0" name="os.family" value="Dell Remote Access Controller"/>
+    <param pos="0" name="os.product" value="Dell Remote Access Controller"/>
+    <param pos="0" name="os.device" value="BMC"/>
+    <param pos="0" name="service.vendor" value="Dell"/>
+    <param pos="0" name="service.family" value="Remote Access Controller"/>
+  </fingerprint>
+
+  <fingerprint pattern="^CN=(?:Sourcefire3D|firepower|ciscoasa),OU=Intrusion Management System,O=(?:Sourcefire\\, Inc.|Cisco Systems\\, Inc),C=US$">
+    <description>Cisco Firepower</description>
+    <example>CN=firepower,OU=Intrusion Management System,O=Cisco Systems\, Inc,C=US</example>
+    <example>CN=ciscoasa,OU=Intrusion Management System,O=Cisco Systems\, Inc,C=US</example>
+    <example>CN=Sourcefire3D,OU=Intrusion Management System,O=Sourcefire\, Inc.,C=US</example>
+    <param pos="0" name="os.vendor" value="Cisco"/>
+    <param pos="0" name="os.family" value="Firepower"/>
+    <param pos="0" name="os.product" value="IPS"/>
+    <param pos="0" name="hw.vendor" value="Cisco"/>
+    <param pos="0" name="hw.family" value="Firepower"/>
+    <param pos="0" name="hw.product" value="Firepower"/>
+    <param pos="0" name="hw.device" value="IPS"/>
+    <param pos="0" name="hw.cpe23" value="cpe:/h:cisco:firepower:-"/>
+  </fingerprint>
+
+  <fingerprint pattern="^CN=ASA Temporary Self Signed Certificate$">
+    <description>Cisco ASA</description>
+    <example>CN=ASA Temporary Self Signed Certificate</example>
+    <param pos="0" name="os.vendor" value="Cisco"/>
+    <param pos="0" name="os.family" value="Adaptive Security Appliance"/>
+    <param pos="0" name="os.product" value="VPN"/>
+    <param pos="0" name="hw.vendor" value="Cisco"/>
+    <param pos="0" name="hw.family" value="Adaptive Security Appliance"/>
+    <param pos="0" name="hw.product" value="Adaptive Security Appliance"/>
```

Comment:
Why is this product different than `os`?

Reply:
This matches the current fingerprints in http_servers.xml (Cisco Aware), as well as the http_cookies.xml entry. It seems weird though. I could use a primer on what these fields should be set to and how os and hw should relate for hardware appliances.

Reply:
It is a tricky area largely because there are lots of bad counterexamples in current recog that need to be cleaned up.  In general, as you might guess, `os` is for anything we can reasonably and usefully say about the operating system and `hw` is for the hardware or physical device itself.